### ISSUE TITLE
raw listener.go appears to only be seeing responses to local clients

### ISSUE THREAD

**Issue by dt** [buger/gor#207]
Was trying out middleware and was never seeing any original response (2) messages.

I'm capturing with `sudo ./gor --input-raw :9991 --output-http http://localhost:9992 --middleware examples/mdleware/echo.sh -verbose` and sending requests from a remote host.

Adding lots of printlns, eventually all the way down to listener.go's `isValidPacket`, I noticed I never saw a packet with `srcPort == 9991`.

At one point I tried listening only on 127.0.0.1 (to reduce the total traffic I was println'ing) and thus switched to making my requests with a local client, and suddenly listener.go *did* see those.
When I went back to listening on `:9991`, but still using a local client, it still saw them. When I went back to making my requests from a remote host I again got nothing.

As a sanity check, I fired up tcpdump and it sees the responses in all cases.

Steps to reproduce:
1) start up two dummy servers on 9991 and 9992 (e.g. `python -m SimpleHTTPServer`).
2) run gor with example echo middleware capturing on 9991:
 `sudo ./gor --input-raw :9991 --output-http http://localhost:9992 --middleware examples/mdleware/echo.sh -verbose` 
3) curl localhost:9991
4) from another host, rerun curl from 3).

**Comment by arothian** [comment]
I've noticed this behavior as well. I have nginx running in front of jboss on a single host. If I listen to nginx's ports I don't see any response data, but if I listen to jboss's port I can see the requests/responses between nginx and jboss.

**Comment by buger** [comment]
Sorry for the delay, does it helps if you explicitly set ip address to --input-raw argument?

**Comment by buger** [comment]
Also seems like this bug happens without middleware as well?

**Comment by buger** [comment]
As temporary solution i think it will work if you specify multiple --input-raw for same gor command but with different ips.

**Comment by buger** [comment]
I reproduced bug, and it definitely damn weird. I event tried Go 1.5, same.

**Comment by arothian** [comment]
@buger  I tried with an address explicity set in input-raw and it still doesn't see the responses (using nginx here). Middleware doesn't seem to be at play here because it happens when I'm using --output-file to record.

**Comment by siemar** [comment]
@buger, was it possible to find out was went wrong with Gor and did you have an idea in which release it can be fixed? Do you have an idea when the next release will be shipped?
We whant to use Gor for an comparison between two application server.
I hope you have good news for me and i thank you for an answer from you.

**Comment by acubens70** [comment]
This issue seems to occur under particular applications. I tested with 2 different applications: At the first application, I could not get HTTP responses from remote client requests but only from local requests. At the second application, I got the HTTP responses both from local and remote client requests.
I tried to analyze the problem, but I couldn't manage it. Under which cicrumstances doesn't Gor receive any outgoing responses while Gor listens by invoking
conn, e := net.ListenPacket("ip4:tcp", t.addr)
at listener.go? Could the be a bug at Go while sniffing the HTTP traffic?
